backend/movement/movement.py

```diff
@@ -0,0 +1,249 @@
+        """
+        self.layout = layout
+        self.animate = animate
+        self.players = {}
+        self.stations = {}
+
+    def _build_players(self, environment_json):
+        """
+        Builds the players in the environment.
+
+        Args:
+            environment_json (dict): The environment json.
+
+        Returns:
+            players (Dictionary[str, Player]): The list of players mapping
+                player name to player object.
+        """
+        players = {}
+        for player in environment_json["players"]:
+            name = player["name"]
+            pos = (player["x"], player["y"])
+            direction = (player["direction"][0], player["direction"][1])
+            players[name] = Player(name, pos, direction)
+        return players
+    
+    def _build_stations(self, environment_json):
+        """
+        Builds the stations in the environment.
+
+        Args:
+            environment_json (dict): The environment json.
+
+        Returns:
+            stations (Dictionary[str, Station]): The list of stations mapping
+                station name to station object.
+        """
+        stations = {}
+        for station in environment_json["stations"]:
+            name = station["name"]
+            pos = (station["x"], station["y"])
+            stations[name] = pos
+        return stations
+    
+    def initialize(self, environment_json):
+        """
+        Initializes the movement.
+
+        Args:
+            animate (bool): Whether or not to enable animate mode.
+
+        Returns:
+            movement (Movement): The movement object.
+        """
+        self.players = self._build_players(environment_json)
+        self.stations = self._build_stations(environment_json)
+
+        return self
+
+    def _get_station_locations(self):
+        """
+        Gets the locations of all stations in the layout.
+        
+        Returns:
+            station_locations (List[tuple]): List of (x, y) positions of stations
+        """
+        station_locations = []
+        for i, row in enumerate(self.layout):
+            for j, col in enumerate(row):
+                if col is not None:
+                    station_locations.append((j, len(self.layout) - i - 1))
+        return station_locations
+    
+    def _get_possible_destinations(self, player, destination):
+        """
+        Gets the possible destination positions for a player.
+
+        Args:
+            player (Player): The player.
+            destination (tuple): The destination position.
+
+        Returns:
+            possible_destinations (List[tuple]): The possible destination positions.
+        """
+        possible_destinations = []
+        other_station_locations = self._get_station_locations()
+        player_locations = [(p.pos[0], p.pos[1]) for p in self.players.values() if p != player]
+        player_destinations = [p.path[-1] for p in self.players.values() if p != player and p.path != []]
```

Comment:
it is odd that the player maintains their path when the path is related to movement. the path a Player takes is not inherent to the Player but rather to Movement. if we forsee that the Player's path will be used by other modules, then it makes sense to leave `path` in Player; otherwise, it makes more sense for it to be maintained by Movement and the Player's position is updated by Movement (position makes sense to be in Player since it is likely to be used elsewhere, e.g. a module that checks if a Player is in range of something).

Reply:
Path now maintained by a class called PlayerData in movement.py